src/types.ts

```diff
@@ -1907,6 +1905,147 @@ export class ZodUnion<T extends ZodUnionOptions> extends ZodType<
   };
 }
 
+/////////////////////////////////////////////////////
+/////////////////////////////////////////////////////
+//////////                                 //////////
+//////////      ZodDiscriminatedUnion      //////////
+//////////                                 //////////
+/////////////////////////////////////////////////////
+/////////////////////////////////////////////////////
+
+type ZodDiscriminatedUnionOption<
```

Comment:
I believe this type should be exported, as all of the types in the package are.
Should I open a PR?
@alexxander